signer/signer.go

```diff
@@ -130,13 +130,17 @@ type Configuration struct {
 	SignerOpts crypto.SignerOpts `json:"signer_opts,omitempty" yaml:"signeropts,omitempty"`
 
 	isHsmAvailable bool
-	hsmCtx         *pkcs11.Ctx
+	Hsm HSM
 }
 
 // InitHSM indicates that an HSM has been initialized
 func (cfg *Configuration) InitHSM(ctx *pkcs11.Ctx) {
 	cfg.isHsmAvailable = true
-	cfg.hsmCtx = ctx
+	cfg.Hsm = &AWSHSM{
+		GenericHSM{
```

Comment:
Needing an explicit `GenericHSM` looks weird to me, and feels like it ought to be unnecessary. My go-fu wasn't strong enough to figure out how to simply set `ctx` directly in `AWSHSM`, however.

Reply:
Ah, yeah, this is def Go preferring explicit over implicit. The recommendation would be to make a newAWSHSM(ctx) func to hide the details. Honestly, I think my feedback would be to do so, anyway.

Reply:
Ah; yes, I've seen that pattern before come to think of it. Will do.